sdcm/cluster_baremetal.py

```diff
@@ -0,0 +1,150 @@
+                                                  ssh_login_info=ssh_login_info,
+                                                  node_prefix=node_prefix)
+
+    @property
+    def public_ip_address(self):
+        return self._public_ip
+
+    @property
+    def private_ip_address(self):
+        return self._private_ip
+
+    def set_hostname(self):
+        self.remoter.run('sudo hostnamectl set-hostname {}'.format(self.name))
+
+    def restart(self):
+        self.remoter.run('sudo reboot -h now', ignore_status=True)
+
+    def destroy(self):
+        # do nothing
+        pass
+
+
+class PhysicalMachineCluster(cluster.BaseCluster):
+
+    def __init__(self, **kwargs):
+        self.nodes = []
+        self.credentials = kwargs.get('credentials')
+        params = kwargs.get('params')
+        n_nodes = kwargs.get('n_nodes')
+        self._node_public_ips = kwargs.get('public_ips', None) or []
+        self._node_private_ips = kwargs.get('private_ips', None) or []
+        node_cnt = n_nodes[0] if isinstance(n_nodes, list) else n_nodes
+        if len(self._node_public_ips) < node_cnt or len(self._node_private_ips) < node_cnt:
+            raise NodeIpsNotConfiguredError('Physical hosts IPs are not configured!')
+        super(PhysicalMachineCluster, self).__init__(node_prefix=kwargs.get('node_prefix'),
+                                                     n_nodes=n_nodes,
+                                                     params=params)
+
+    def _create_node(self, name, public_ip, private_ip):
+        return PhysicalMachineNode(name,
+                                   public_ip,
+                                   private_ip,
+                                   credentials=self.credentials[0],
+                                   base_logdir=self.logdir,
+                                   node_prefix=self.node_prefix)
+
+    def add_nodes(self, count, dc_idx=0):
+        for node_index in xrange(count):
+            node_name = '%s-%s' % (self.node_prefix, node_index)
+            self.nodes.append(self._create_node(node_name,
+                                                self._node_public_ips[node_index],
+                                                self._node_private_ips[node_index]))
+
+    def destroy(self):
+        logger.info('Destroy nodes')
+        for node in self.nodes:
+            node.destroy(force=True)
+
+
+class ScyllaPhysicalCluster(PhysicalMachineCluster, cluster.BaseScyllaCluster):
+
+    def __init__(self, **kwargs):
+        self._user_prefix = kwargs.get('user_prefix', cluster.DEFAULT_USER_PREFIX)
+        self._node_prefix = '%s-%s' % (self._user_prefix, BASE_NAME)
+        super(ScyllaPhysicalCluster, self).__init__(node_prefix=kwargs.get('node_prefix', self._node_prefix),
+                                                    **kwargs)
+        self.seed_nodes_private_ips = None
+        self._seed_node_rebooted = False
+
+    def node_setup(self, node, verbose=False):
```

Comment:
Do we need to support to connect existing cluster? Installation & setup can be skipped.

@roydahan

Reply:
I think we can add support for an existing environment is some common place to support all the types. + see my previous comment

Reply:
@amoskong, see my comments above.
In general we want to support it, but lower priority for now.

This PR will be needed for Power/arm testing.
We should get machines for them outside our current supported backends.